server/showcase.go

```diff
@@ -15,40 +15,47 @@
 package server
 
 import (
+	"context"
 	"fmt"
 	"io"
 	"strconv"
 	"strings"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/empty"
 	pb "github.com/googleapis/gapic-showcase/server/genproto"
-	"github.com/grpc/grpc-go/status"
-
-	"golang.org/x/net/context"
-
 	lropb "google.golang.org/genproto/googleapis/longrunning"
 	statuspb "google.golang.org/genproto/googleapis/rpc/status"
 	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
+type uniqID struct {
+	i int64
+}
+
+func (u *uniqID) id() int64 {
+	return atomic.AddInt64(&u.i, 1) - 1
+}
+
 // NewShowcaseServer returns a new ShowcaseServer for the Showcase API.
 func NewShowcaseServer(opStore OperationStore) pb.ShowcaseServer {
 	return &showcaseServerImpl{
 		operationStore: opStore,
-		nowF:           time.Now,
 		sleepF:         time.Sleep,
 	}
 }
 
 type showcaseServerImpl struct {
+	uid uniqID
+
+	mu             sync.Mutex
```

Comment:
The mutex is idiomatically at the top of the "block" containing things it's protecting.